tensorflow/core/kernels/mkl_fused_batch_norm_op_test.cc

```diff
@@ -0,0 +1,259 @@
+      TF_EXPECT_OK(NodeDefBuilder("MklFusedBatchNorm", "_MklFusedBatchNorm")
+                       .Input(FakeInput(dtype))
+                       .Input(FakeInput(DT_FLOAT))
+                       .Input(FakeInput(DT_FLOAT))
+                       .Input(FakeInput(DT_FLOAT))
+                       .Input(FakeInput(DT_FLOAT))
+                       .Input(FakeInput(DT_UINT8))
+                       .Input(FakeInput(DT_UINT8))
+                       .Input(FakeInput(DT_UINT8))
+                       .Input(FakeInput(DT_UINT8))
+                       .Input(FakeInput(DT_UINT8))
+                       .Attr("exponential_avg_factor", exponential_avg_factor)
+                       .Attr("epsilon", 0.001)
+                       .Attr("is_training", is_training)
+                       .Attr("_kernel", "MklLayoutDependentOp")
+                       .Finalize(node_def()));
+      TF_EXPECT_OK(InitOp());
+
+      AddInputFromArray<float>(input.shape(), input.flat<T>());
+      AddInputFromArray<float>(scale.shape(), scale.flat<T>());
+      AddInputFromArray<float>(offset.shape(), offset.flat<T>());
+      AddInputFromArray<float>(mean.shape(), mean.flat<T>());
+      AddInputFromArray<float>(variance.shape(), variance.flat<T>());
+      for (int i = 0; i < 5; ++i)
+        AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+      TF_ASSERT_OK(RunOpKernel());
+
+      CommonTestUtilities<T> test_util;
+      test_util.PerformConversion(dtype, *GetOutput(0), *GetOutput(5), output);
+
+      CommonTestUtilities<T> test_util_mean;
+      test_util_mean.PerformConversion(dtype, *GetOutput(1), *GetOutput(6),
+                                       batch_mean);
+
+      CommonTestUtilities<T> test_util_var;
+      test_util_var.PerformConversion(dtype, *GetOutput(2), *GetOutput(7),
+                                      batch_var);
+    };
+
+    CommonTestUtilities<T>::VerifyTensorsClose(exponential_avg_factor,
+                                               is_training, run, run_mkl);
+  }
+};
+
+TYPED_TEST_CASE_P(FusedBatchNormOpTest);
+
+TYPED_TEST_P(FusedBatchNormOpTest, Training) {
+  const float exponential_avg_factor = 1.0;
+  const bool is_training = true;
+  this->VerifyFusedBatchNorm(exponential_avg_factor, is_training);
+}
+
+TYPED_TEST_P(FusedBatchNormOpTest, TrainingRunningMean) {
+  const float exponential_avg_factor = 0.5;
+  const bool is_training = true;
+  this->VerifyFusedBatchNorm(exponential_avg_factor, is_training);
+}
+
+TYPED_TEST_P(FusedBatchNormOpTest, Inference) {
+  const float exponential_avg_factor = 1.0;
+  const bool is_training = false;
+  this->VerifyFusedBatchNorm(exponential_avg_factor, is_training);
+}
+
+TYPED_TEST_P(FusedBatchNormOpTest, InferenceIgnoreAvgFactor) {
+  const float exponential_avg_factor = 0.5;
+  const bool is_training = false;
+  this->VerifyFusedBatchNorm(exponential_avg_factor, is_training);
+}
+
+REGISTER_TYPED_TEST_CASE_P(FusedBatchNormOpTest, Training, TrainingRunningMean,
```

Comment:
`REGISTER_TYPED_TEST_CASE_P` has be [deprecated](https://github.com/google/googletest/blob/e588eb1ff9ff6598666279b737b27f983156ad85/googletest/include/gtest/gtest-typed-test.h#L302-L308). Please use [REGISTER_TYPED_TEST_SUITE_P]( https://github.com/google/googletest/blob/e588eb1ff9ff6598666279b737b27f983156ad85/googletest/include/gtest/gtest-typed-test.h#L147) instead.